### ISSUE TITLE
Bug: lexical is triggering all my accessibility checks

### ISSUE THREAD

**Issue by sneko** [facebook/lexical#3779]
Hi,

I chose Lexical because on Reddit and on the Github page it was announced as accessible. Unfortunately it's far from that, my project has:
* `axe-core` to check DOM structure
* `jsx-a11y` as an `eslint` plugin

Even with no content in my editor I get dozens (hundreds?) of accessibility violations. When I tried to silent some rules others come... Note that before adding `lexical` my project triggers no error and that I did not silent any rule.

To reproduce, just take the playground folder and enable those tools.

EDIT: note that in the playground folder I saw tons of `eslint-disable-next-line XXXXXX`, the issue with that is it makes things not accessible, and that when we run on our end the checks it panics because it's really hard to silent rule on specific nested elements.

**Comment by trueadm** [comment]
The core of Lexical is very accessible – this is what we use at Facebook and have had audited. Yes, there are some known issues around JAWS, Dragon etc, but these isolated issues mostly. When we talk about accessibility, we talk about the core typing/selection functionality of the editor, and not about the various fancy plugins or toolbars/tooltips – they're not Lexical, that's all React really in terms of the playground.

The playground is a kitchen sink of things, many of which have been contributed by others. As such, there's likely to be things like eslint ignores etc around the codebase and things that aren't quite right. The playground is a big contributed space where people can showcase various efforts and also contribute. If you'd like to address some of the issues you've found – any help would be much appreciated.

**Comment by acywatson** [comment]
yes, this is a fundamental misunderstanding of what Lexical is and how it works. If you can identify specific accessibility issues with the core library (or really any public API, which the playground is most certainly not) then we can have a chat about that.

**Comment by sneko** [comment]
Sorry for the context misunderstanding!

Question : after some research I did not find an all-in-one React library to set up a lexical with plugins+toolbar (I mean like 1-2 imports and let's go). I just found some copy/paste of the playground. Do you know some?

Since then, I wanted to benefit from my UI theming (dark mode but also components appearance...), I moved the playground to the MUI design system. Don't know if it can help someone but here it is:
https://github.com/inclusion-numerique/mediature/tree/dev/packages/ui/src/Editor

<img width="905" alt="Screenshot 2023-01-27 at 12 38 17" src="https://user-images.githubusercontent.com/8195958/215092456-70aa0826-eef0-4085-847c-6d062aa1699f.png">

It's far from being library ready, things are still harcoded (some colors where MUI didn't do the job) and I adjusted text language to french for my immediate use case. If someone looks into it you will need to get your hands dirty :D 

I kept plugins I wanted and removed the UI possibility to customize the color, otherwise it's too complicated when the content should be displayed on light/dark website, but also when sending the content over emails. @trueadm do you know a way to prevent color customization? (I just removed it from the menu... but it does not prevent it if copy/paste text from elsewhere. Ideally I would like that when validating the `editorState` it's also prevented if a custom color has been set (the editor should have forced the "idle" one).

Thank you for your work,
